lib/util/http/host-rules.ts

```diff
@@ -217,16 +217,36 @@ export function applyHostRule<GotOptions extends HostRulesGotOptions>(
   return options;
 }
 
-export function getConcurrentRequestsLimit(url: string): number | null {
+export function getConcurrentRequestsLimit(
+  url: string,
+  defaultValue: number | null,
+): number | null {
   const { concurrentRequestLimit } = hostRules.find({ url });
-  return is.number(concurrentRequestLimit) && concurrentRequestLimit > 0
-    ? concurrentRequestLimit
-    : null;
+
+  if (!is.number(concurrentRequestLimit)) {
```

Comment:
we need to check and update docs. setting those values to null is like setting to default of `5`.
to disable limit set to `-1`